### ISSUE TITLE
Pkg activate fails to load the correct versions of the packages

### ISSUE THREAD

**Issue by jbrea** [fonsp/Pluto.jl#1511]
I am experiencing a very annoying and difficult-to-debug behaviour when updating to 0.16.0 or 0.16.1:
In [my course](https://github.com/jbrea/MLCourse) I do not use the Pluto pkg manager but ship a custom Manifest file.
Each notebook starts with `using Pkg; Pkg.activate("MLCourse")`. Until Pluto version 0.15.1 this works fine.
With the new Pluto versions it seems other versions of the packages are loaded.
The weird thing is: `Pkg.status()` from within the Pluto notebook shows the correct versions (in the REPL),
but `@which somefunction()` points me to other versions of the packages, e.g. [in this notebook](https://github.com/jbrea/MLCourse/blob/main/notebooks/supervised_learning.jl) I expect version v1.1.0 of MLJOpenML to be loaded but `@which MLJOpenML(554)` points to v1.0.0 of MLJOpenML.

With the pop-up saying that there is a new version of Pluto the students are really tempted to update their Pluto version, just to find that after the update the notebooks don't work anymore...

**Comment by fonsp** [comment]
MLJOpenML.jl is not in your [Project.toml](https://github.com/jbrea/MLCourse/blob/bio322fall21/Project.toml) (nor Manifest.toml), which means that it is being loaded from your global environment. 

This is because of [environment stacking](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks). If you do `Pkg.activate(some_dir)`, then your stack will be `["some_dir", "@v1.6", "@stdlib"]`. When loading a package, Julia will go through these environments one-by-one until the package is found.

Fix it by adding MLJOpenML to your project. To find bugs like this in the future, I recommend that you disable environment stacking by removing `@v1.6` from your load path. I did this in one notebook in our course: https://github.com/mitmath/18S191/blob/2656801ec8bcea079acbd8c1b69b5e97dba1a1db/notebooks/week3/transformations2.jl#L46-L48

---

Is the problem this happening after this?

**Comment by fonsp** [comment]
Out of interest: why are you not using the built-in package manager?

**Comment by jbrea** [comment]
> MLJOpenML.jl is not in your Project.toml (nor Manifest.toml), which means that it is being loaded from your global environment.

No, this is not the problem (if it where it would also show up in Pluto v0.15.1 I guess). I demonstrated the problem with a notebook on branch main. MLJOpenML [is in the Manifest](https://github.com/jbrea/MLCourse/blob/main/Manifest.toml#L863) on branch main.

> Out of interest: why are you not using the built-in package manager?

I found it easier to have a single Manifest file for the whole course, install all dependencies once and just activate this environment when opening the notebooks instead of having to care about the dependency versions of each individual notebook. I also thought that it would be easier to update packages, because I would do it once for the whole environment and not for each notebook separately. Or is there an easy way to keep the "Manifests" of several notebooks in sync?

**Comment by jbrea** [comment]
Some more information:
1. Running the notebook (supervised_learning.jl on branch main) in the repl loads the correct versions of the packages.
2. `Pluto.run(notebook = "supervised_learning.jl")` loads the wrong versions with v0.16.0 and v0.16.1 but it work on v0.15.1

I have difficulties to further debug this. Any ideas?

**Comment by fonsp** [comment]
Any idea where it is getting 1.0.0 from? The global env? Can you try adding my LOAD_PATH patch (linked above) and see if that fixes it?

**Comment by jbrea** [comment]
Ah, sorry, I forgot to report: the LOAD_PATH patch didn't work. With `LOAD_PATH = ["@", "@stdlib"]` I still get the same error.
But I think it does nevertheless get the package from the `@v1.6` environment, because it loads the same version as the one defined in `.julia/environments/v1.6/Manifest.toml`. It loads the correct version if MLJOpenML is not defined in that Manifest file.

**Comment by juliohm** [comment]
The new patch 1.6.2 of Pluto is now breaking my previously working notebook with DrWatson.@quickactivate. Can you please clarify what is the expected usage of activate when the user is not interested in the built-in package manager?

We were using

```julia
begin
  using DrWatson
  @quickactivate "MyClass"
end

begin
  using Foo
  using Bar
end
```

cc: @Datseris

**Comment by Datseris** [comment]
Great, thanks for the great work! I'll put a note in the docs by tonight!!!
